regnety/dataset/imagenet.py

```diff
@@ -80,159 +80,145 @@ def decode_example(self, example: tf.Tensor) -> dict:
             "synset": synset,
         }
 
+
     def _read_tfrecs(self) -> Type[tf.data.Dataset]:
         """Function for reading and loading TFRecords into a tf.data.Dataset.
-
         Returns:
             A tf.data.Dataset
         """
-        ds = tf.data.TFRecordDataset(self.tfrecs_filepath)
-        ds = ds.map(
-            lambda example: tf.io.parse_example(example, _TFRECS_FORMAT),
-            num_parallel_calls = tf.data.AUTOTUNE
-        )
+
+        files = tf.data.Dataset.list_files(self.tfrecs_filepath)
+
+        options = tf.data.Options()
+        options.experimental_deterministic = False
+
+        files = files.with_options(options)
```

Comment:
We can group these. 

Also, when using this and `interleave()` together, does that collectively boost performance? Have you tried using them separately and observed any significant changes?

Reply:
If you are referring to the usage of `tf.data.Options()`, then no, I didn't see any substantial improvements. But using interleave over plain TFRecordDataset does give some improvements. 

I'll delete the `options` code, as we are specifying deterministic = False in interleave()

Reply:
Cool. That sounds reasonable.